src/Driver/AbstractOracleDriver/EasyConnectString.php

```diff
@@ -70,14 +70,21 @@ public static function fromConnectionParameters(array $params): self
             $connectData['INSTANCE_NAME'] = $params['instancename'];
         }
 
+        $protocol = 'TCP';
+        if (isset($params['protocol'])) {
```

Comment:
I believe we should stop putting driver-specific parameters into the driver-agnostic `$params`. Please try moving it to driver options.

Reply:
Hey @morozov - It's an Oracle parameter rather than a driver-specific function. You'll find it used when you generate a connection string from within OCI Cloud.  

It's not dissimilar to how it's done in java - as much as it's driven within an oracle namespace. 

Where else would you suggest I put this if not in an AbstractOracleDriver/EasyConnectString.php ? 

Also noting that 'TCP' as the protocol is already set statically in this portion of code? 

Cheers!

Reply:
> It's an Oracle parameter rather than a driver-specific function.

So… it's a parameter that we only ever need when connecting through the Oracle drivers which makes it very driver-specific.

Reply:
Hi @derrabus - and the change is in AbstractOracleDriver - am I missing something?

Reply:
As @morozov said, `$params` is driver-agnostic. That's regardless of which class receives it.

Reply:
Ah now I follow what was being said. 

I’ll look at the other drivers to see how specific database settings are made available

Reply:
@greg0ire @derrabus @morozov I have moved the configuration item into driverOptions, in line with the 'exclusive' configuration item that already existed. Hopefully, that's better.